sormas-backend/src/main/java/de/symeda/sormas/backend/common/StartupShutdownService.java

```diff
@@ -811,6 +820,41 @@ private void upgrade() {
 					userRoleService.ensurePersisted(userRole);
 				}
 				break;
+			case 478:
+				// Add proper captions and translations to occupation types
+				List<OccupationType> occupationTypes = customizableEnumFacade.getEnumValues(CustomizableEnumType.OCCUPATION_TYPE, null);
+				occupationTypes.forEach(o -> {
+					String caption = I18nProperties.getEnumCaption(null, OccupationType.I18N_PREFIX, o.getValue());
+					List<String> translations = new ArrayList<>();
+					Arrays.stream(Language.values()).forEach(l -> {
+						translations.add(
+							String.format(
+								"{\"languageCode\":\"%s\",\"value\":\"%s\"}",
+								l.getLocale(),
+								I18nProperties.getEnumCaption(l, OccupationType.I18N_PREFIX, o.getValue())));
+					});
+					String translationsString = "[" + String.join(",", translations) + "]";
+					String propertiesString = "{\"hasDetails\":true}";
```

Comment:
Yeah, that's done on purpose. The alternative would be to do database queries for each of the occupation types to find out whether the details field has been filled for any of them; that could be quite costly, and I'm not sure whether it's worth it or whether it's okay to just say we enable the details field for all of them and write in the release notes that server admins need to turn the details field off for those occupation types that they don't want to use it for.

Even if the details field is NOT filled in for an occupation type, we can't be sure that it's not supposed to be used anyway because maybe users just didn't complete the data.

So the behaviour that you observed is correct according to my current implementation; the question therefore is whether this implementation makes sense or whether we need to change it. What's your impression after reading my comment? Is there an alternative solution you'd like to suggest?

Reply:
since the behavior of Occupation_Details field is hardcoded in PersonEditForm then I believe we should not find details values anywhere else than in those occupations mentioned in the Switch. I think this Switch could be hardcoded in the upgrade method from StartupShutdownService for "propertiesString" variable.